### PR TITLE
fix: migrate from tap to node test and c8

### LINKED ISSUE #129: Migrate to `node:test` and `c8`
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the issue has not already been raised

### Issue

_No response_

### PR DESCRIPTION
**Pull request by dancastillo**
#### Checklist

Closes #129 

This PR migrates from `tap` to `node:test` and `c8`

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by climba03003 — CHANGES_REQUESTED** [review]

File: test/index.test.js [hunk 1]
```diff
@@ -1,184 +1,163 @@
...
 
-const { test } = require('tap')
+const { test } = require('node:test')
+const assert = require('node:assert')
 const createError = require('..')
 
-test('Create error with zero parameter', t => {
-  t.plan(6)
-
+test('Create error with zero parameter', (t) => {
```
[the remark this answers is not in the document]
Comment (jsumners) [reply test/index.test.js:7]: You cannot use `t.assert` as it is not available in Node.js 18 or 20.

File: test/index.test.js [hunk 2]
```diff
@@ -1,184 +1,163 @@
 'use strict'
 
-const { test } = require('tap')
+const { test } = require('node:test')
```
Comment (jsumners) [inline test/index.test.js:3]:
```suggestion
const test = require('node:test')
```

File: test/index.test.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (climba03003) [reply test/index.test.js:7]: It is available on node@20, but not node@18.
That's why I request to update using v5 workflow

[the remark this answers is not in the document]
Comment (climba03003) [reply test/index.test.js:7]: See https://github.com/nodejs/node/releases/tag/v20.15.0

File: test/index.test.js [hunk 3]
```diff
@@ -1,184 +1,194 @@
...
+  assert.throws(
+    () => createError('code'),
+    new Error('Fastify error message must not be empty')
+  )
 })
 
-test('Create error with different base', t => {
-  t.plan(7)
+test('Create error with different base', (t) => {
+  t.plan(6)
```
Comment (climba03003) [inline test/index.test.js:129]:
```suggestion
  t.plan(7)
```

**Review by climba03003 — CHANGES_REQUESTED** [review]

File: test/index.test.js (hunk 3 above)
Comment (dancastillo) [reply test/index.test.js]: Should be fixed now

**Review by jsumners — APPROVED** [review]

**Review by climba03003 — APPROVED** [review]

**Comment by gurgunday** [comment]
Thank you @dancastillo!

**Pull request merged**